pkg/operator/controller/ingress/controller.go

```diff
@@ -1205,3 +1220,42 @@ func (r *reconciler) allRouterPodsDeleted(ingress *operatorv1.IngressController)
 
 	return true, nil
 }
+
+// updateInfraIngressLoadBalancerIPs updates GCP's PlatformStatus with Ingress LB IPs when the DNSType is `ClusterHosted`.
+func updateInfraIngressLoadBalancerIPs(service *corev1.Service, infraConfig *configv1.Infrastructure) (bool, error) {
+	platformStatus := infraConfig.Status.PlatformStatus
+	if platformStatus != nil {
+		switch platformStatus.Type {
+		case configv1.GCPPlatformType:
+			if platformStatus.GCP != nil && platformStatus.GCP.CloudLoadBalancerConfig != nil && platformStatus.GCP.CloudLoadBalancerConfig.DNSType == configv1.ClusterHostedDNSType {
+				// The cluster has to run its own CoreDNS pod for DNS. Update Infra CR
+				// with the Ingress LB IPs. These values are used to configure the
+				// in-cluster DNS to provide resolution for *.apps.
+				if infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted == nil {
+					infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted = &configv1.CloudLoadBalancerIPs{}
+				}
+				ingresses := service.Status.LoadBalancer.Ingress
+				ingressLBIPs := []configv1.IP{}
+				for _, ingress := range ingresses {
+					if len(ingress.IP) > 0 {
+						ingressLBIPs = append(ingressLBIPs, configv1.IP(ingress.IP))
+					}
+				}
+				ipCmpOpts := []cmp.Option{
+					cmpopts.EquateEmpty(),
+					cmpopts.SortSlices(func(a, b configv1.IP) bool {
+						return a < b
+					}),
+				}
+				if !cmp.Equal(infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs, ingressLBIPs, ipCmpOpts...) {
+					infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs = ingressLBIPs
+					return true, nil
+				}
+			}
+			return false, nil
+		default:
+			return false, nil
+		}
+	}
+	return false, fmt.Errorf("invalid PlatformStatus within Infrastructure config")
```

Comment:
Instead of the following:
```go
	platformStatus := infraConfig.Status.PlatformStatus
	if platformStatus != nil {
		switch platformStatus.Type {
		case configv1.GCPPlatformType:
			// ...
			return false, nil
		default:
			return false, nil
		}
	}
	return false, fmt.Errorf("invalid PlatformStatus within Infrastructure config")
```
Please do the following:
```go
	platformStatus := infraConfig.Status.PlatformStatus
	if platformStatus == nil {
		return false, fmt.Errorf("invalid PlatformStatus within Infrastructure config")
	}
	switch platformStatus.Type {
	case configv1.GCPPlatformType:
		// ...
		return false, nil
	default:
		return false, nil
	}
```
This puts the error condition (`platformStatus == nil`) with the error value (`fmt.Errorf("invalid PlatformStatus within Infrastructure config")`).  As a nice side-effect, it reduces the function's main logic.

Reply:
Done